quarkScriptAgent.py

```diff
@@ -0,0 +1,144 @@
+import os
+import re
+
+from termcolor import colored
+
+from langchain_openai import ChatOpenAI
+from langchain.agents import tool, AgentExecutor
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain.agents.output_parsers.openai_tools import OpenAIToolsAgentOutputParser
+from langchain.agents.format_scratchpad.openai_tools import (
+    format_to_openai_tool_messages,
+)
+
+from quark.script import Rule, _getQuark, QuarkResult
+
+if "OPENAI_API_KEY" not in os.environ:
+    api_key = input("OpenAI API Key: ")
+    os.environ["OPENAI_API_KEY"] = api_key
+
+
+@tool
+def loadRule(rulePath):
```

Comment:
Please add a type hint for `rulePath`.
```python
def loadRule(rulePath: str):
```


Without the type hint, the LLM can not pass the rule path to this tool. Here is the error output.

```
User Input: Initialize the rule instance with the rule path set to "rule.json"

Agent:  The rule has been successfully initialized with the path set to "rule.json". What would you like to do next?

User Input: Run Quark Analysis using the rule instance on the apk sample "ovaa.apk",             and Check if the parameters are hard-coded. If yes, display the hard-coded values.
Traceback (most recent call last):
  File "/Users/shengfeng/codespace/quark-script/quarkScriptAgent.py", line 137, in <module>
    response = agent_executor.invoke({
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/chains/base.py", line 166, in invoke
    raise e
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/chains/base.py", line 156, in invoke
    self._call(inputs, run_manager=run_manager)
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/agents/agent.py", line 1612, in _call
    next_step_output = self._take_next_step(
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/agents/agent.py", line 1318, in _take_next_step
    [
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/agents/agent.py", line 1318, in <listcomp>
    [
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/agents/agent.py", line 1403, in _iter_next_step
    yield self._perform_agent_action(
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain/agents/agent.py", line 1425, in _perform_agent_action
    observation = tool.run(
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain_core/tools.py", line 634, in run
    raise error_to_raise
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain_core/tools.py", line 607, in run
    response = context.run(self._run, *tool_args, **tool_kwargs)
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/langchain_core/tools.py", line 944, in _run
    return self.func(*args, **kwargs)
  File "/Users/shengfeng/codespace/quark-script/quarkScriptAgent.py", line 29, in loadRule
    ruleInstance = Rule(rulePath)
  File "/Users/shengfeng/codespace/quark-script/.venv/lib/python3.9/site-packages/quark/core/struct/ruleobject.py", line 38, in __init__
    with open(ruleJson) as json_file:
FileNotFoundError: [Errno 2] No such file or directory: 'path/to/rule'
```